cpp/dolfinx/mesh/CMakeLists.txt

```diff
@@ -4,9 +4,12 @@ set(HEADERS_mesh
   ${CMAKE_CURRENT_SOURCE_DIR}/Geometry.h
   ${CMAKE_CURRENT_SOURCE_DIR}/Topology.h
   ${CMAKE_CURRENT_SOURCE_DIR}/MeshTags.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/box_mesh.h
```

Comment:
Use the underscore in the header name is inconsistent with the style we use for all other headers.

I would suggest putting all 'generator' functions in a single header. The multiple files is legacy.